### PR TITLE
indexer easy: migration method gatekeeper config

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

construct cmd line config for method migration gatekeeper, also flip the default logic to better map the progress from empty array.

## Test Plan 

CI

### CODE REVIEW

File: crates/sui-indexer/src/lib.rs [hunk 1]
```diff
@@ -64,6 +64,8 @@ pub struct IndexerConfig {
     pub rpc_server_url: String,
     #[clap(long, default_value = "9000", global = true)]
     pub rpc_server_port: u16,
+    #[clap(long, multiple_occurrences = false, multiple_values = true)]
+    pub migrated_methods: Vec<String>,
```
Comment (wlmyng) [inline crates/sui-indexer/src/lib.rs:68]: question: sorry missing some context here, what are `migrated_methods`?

File: crates/sui-indexer/src/apis/read_api.rs [hunk 2]
```diff
@@ -289,8 +288,8 @@ where
     }
 
     async fn get_total_transaction_number(&self) -> RpcResult<u64> {
-        if self
-            .method_to_be_forwarded
+        if !self
```
Comment (wlmyng) [inline crates/sui-indexer/src/apis/read_api.rs:291]: so if `self.migrated_methods` true then use own api, otherwise use fullnode

**Review by wlmyng — APPROVED** [review]
:makes sense to me:

File: crates/sui-indexer/src/lib.rs (hunk 1 above)
Comment (gegaowp) [reply crates/sui-indexer/src/lib.rs:68]: migrated method uses indexer native impl. while others are forwarded to FN.

we want to validate data returned by FN / indexer RPC endpoints, when the data validated, we can explicitly add endpoints to migrated methods via ops change

**Pull request merged**